packages/babel-core/src/config/item.js

```diff
@@ -64,6 +64,11 @@ class ConfigItem {
    */
   _descriptor: UnloadedDescriptor;
 
+  /**
+   * Used to detect ConfigItem instances from other Babel instances.
+   */
+  _isConfigItem: true;
```

Comment:
```suggestion
  [Symbol.for("@babel/core@7 - ConfigItem")]: true;
```

And we can remove the `instanceof` check and only rely on this.